lib/opentelemetry/trace/exporter/circuit.lua

```diff
@@ -0,0 +1,77 @@
+local util = require("opentelemetry.util")
+local otel_global =  require("opentelemetry.global")
+
+local _M = {
+    OPEN = "open",
+    CLOSED = "closed",
+    HALF_OPEN = "half-open"
+}
+
+local mt = {
+    __index = _M
+}
+
+
+function _M.new(options)
+    options = options or {}
+    local self = {
+        circuit_reset_timeout_ms = options["circuit_reset_timeout_ms"] or 5000,
+        failure_threshold = options["failure_threshold"] or 5,
+        failure_count = 0,
+        open_start_time_ms = nil,
+        state = "closed",
+    }
+    return setmetatable(self, mt)
+end
+
+function _M.should_make_request(self)
+    if self.state == self.CLOSED then
+        return true
+    end
+
+    if self.state == self.OPEN then
+        if (util.gettimeofday_ms() - self.open_start_time_ms) > self.circuit_reset_timeout_ms then
+            self.state = self.HALF_OPEN
+            self.open_start_time_ms = nil
+        else
+            return false
+        end
+    end
+
+    if self.state == self.HALF_OPEN then
```

Comment:
I think we can just return true here, then remove the if condition and log.

Reply:
I removed the conditional, but kept the log. If you feel strongly, I can remove the log. I think it would be helpful when debugging.

7eb4969c688ecc8b9b1390776cccda20af91f73c